local-volume/provisioner/pkg/common/common.go

```diff
@@ -130,20 +144,12 @@ func CreateLocalPVSpec(config *LocalPVConfig) *v1.PersistentVolume {
 // GetVolumeConfigFromMountedConfigMap gets volume configuration from the mounted configmap volume,
 func GetVolumeConfigFromMountedConfigMap() (map[string]MountConfig, error) {
 	storageClassMapping := make(map[string]string)
-	files, err := ioutil.ReadDir(ProvisionerConfigPath)
+	fileContents, err := ioutil.ReadFile(path.Join(ProvisionerConfigPath, ProvisonerStorageClassConfig))
```

Comment:
Here's a different thought I had on doing this:

1. Make ProvisionerConfiguration also with the yaml type.
2. In this function, loop through all the files, and construct a yaml map string with key = filename, value = fileContents.
3. Then you can just unmarshal the entire map into the ProvisionerConfiguration.

Reply:
I think pluggable parses approach is more flexible, because:
1. If there is a problem with config key (some information if wrong or incomplete) in case of bulk unmarshal, everything will fail and we cannot find out which config key caused it. On the other hand per key func parser can deal with unmarshal failures more gracefully.
2. pluggable parser also allows additional config processing, example if some config information is missing, it can check and apply default values. 

would you agree with these advantages?

Reply:
1. is the parsing error not smart enough to say things like "unrecognized field 'foo'" or something like that?
2. We can just have some validation method at the end

I don't envision the config to be really big, so maybe pluggable is a little overkill.

Reply:
map[string]string cannot be used as an input to unmarshal, it has to be []byte. When I pass, combined into one string map, unmarshal is not capable to parse and fill provisioner configuration struct.